src/app/modules/main-module/components/main/main.component.ts

```diff
@@ -152,9 +152,23 @@ export class MainComponent implements OnInit {
   }
 
   checkWin() {
+    this.attempstCounter++;
+    let winValue = true;
     this.wordStatus.forEach((value) => {
-      if (value != 'MATCHED') this.winValue = false;
+      if (value != 'MATCHED') winValue = false;
     });
-    if (this.winValue) this.dialog.open(DialogWinComponent);
+    if (winValue) {
```

Comment:
aquí podéis refactorizar, estáis usando mas o menos la misma funcionalidad para ganar o perder...pensad una posible solución para repetir menos código y para incluir menos if else if